variables.tf

```diff
@@ -66,104 +54,171 @@ variable "domain_name" {
-  default     = "15Gi"
+data "vault_generic_secret" "influxdb_telegraf" {
+  path = "${local.vault_root}/influxdb_telegraf"
 }
 
 variable "influxdb_telegraf_pass" {
-  description = "InfluxDB password for the telegraf user."
+  description = "InfluxDB password for the telegraf user. (default: vault)"
+  default     = ""
 }
 
-variable "prometheus_oauth_github_org" {
-  description = "limit access to prometheus dashboard to members of this org"
+data "vault_generic_secret" "prometheus_oauth" {
+  path = "${local.vault_root}/prometheus_oauth"
 }
 
 variable "prometheus_oauth_client_id" {
-  description = "github oauth client id"
+  description = "github oauth client id. (default: vault)"
+  default     = ""
 }
 
 variable "prometheus_oauth_client_secret" {
-  description = "github oauth client secret"
+  description = "github oauth client secret. (default: vault)"
+  default     = ""
+}
+
+variable "prometheus_oauth_github_org" {
+  description = "limit access to prometheus dashboard to members of this org"
+  default     = "lsst-sqre"
+}
+
+data "vault_generic_secret" "tls" {
+  path = "${local.vault_root}/tls"
+}
+
+variable "tls_crt" {
+  description = "wildcard tls certificate. (default: vault)"
+  default     = ""
+}
+
+variable "tls_key" {
+  description = "wildcard tls private key. (default: vault)"
+  default     = ""
+}
+
+locals {
+  # Name of google cloud container cluster to deploy into
+  gke_cluster_name = "${var.deploy_name}-${var.env_name}"
+
+  # remove "<env>-" prefix for production
+  dns_prefix = "${replace("${var.env_name}-", "prod-", "")}"
+
+  prometheus_k8s_namespace     = "monitoring"
+  kafka_k8s_namespace          = "kafka"
+  grafana_k8s_namespace        = "grafana"
+  influxdb_k8s_namespace       = "influxdb"
+  telegraf_k8s_namespace       = "telegraf"
+  nginx_ingress_k8s_namespace  = "nginx-ingress"
+  kafka_efd_apps_k8s_namespace = "kafka-efd-apps"
+
+  grafana_admin_pass = "${random_string.grafana_admin_pass.result}"
+  grafana_admin_user = "admin"
+
+  vault_root = "secret/dm/square/${var.deploy_name}/${var.env_name}"
+
+  github       = "${data.vault_generic_secret.github.data}"
+  github_user  = "${var.github_user != "" ? var.github_user: local.github["user"]}"
+  github_token = "${var.github_token != "" ? var.github_token: local.github["token"]}"
+
+  grafana_oauth               = "${data.vault_generic_secret.grafana_oauth.data}"
+  grafana_oauth_client_id     = "${var.grafana_oauth_client_id != "" ? var.grafana_oauth_client_id : local.grafana_oauth["client_id"]}"
+  grafana_oauth_client_secret = "${var.grafana_oauth_client_secret != "" ? var.grafana_oauth_client_secret : local.grafana_oauth["client_secret"]}"
+
+  influxdb_admin      = "${data.vault_generic_secret.influxdb_admin.data}"
+  influxdb_admin_pass = "${var.influxdb_admin_pass != "" ? var.influxdb_admin_pass : local.influxdb_admin["pass"]}"
```

Comment:
Will this work?  where are the variables `var.*` defined?

Reply:
The var is declared above in this file.